builder.go

```diff
@@ -332,20 +332,7 @@ func ParseFile(path string) (*parser.Node, error) {
 
 // Step creates a new step from the current state.
 func (b *Builder) Step() *Step {
-	argsMap := make(map[string]string)
-	for _, argsVal := range b.Arguments() {
-		val := strings.SplitN(argsVal, "=", 2)
-		if len(val) > 1 {
-			argsMap[val[0]] = val[1]
-		}
-	}
-
-	userArgs := makeUserArgs(b.Env, argsMap) 
-	dst := make([]string, len(userArgs)+len(b.RunConfig.Env))
-	copy(dst, userArgs)
-	dst = append(dst, b.RunConfig.Env...)
-
-	return &Step{Env: dst}
+	return &Step{Env: mergeEnv(b.Arguments(), mergeEnv(b.Env, b.RunConfig.Env))}
```

Comment:
This is really slick.  Could we add a code comment here talking about this ordering being important so that the values in b.Arguments() won't be added to the list if the key was already created via values found in the ENV fields?

Reply:
Added comment `Include build arguments in the table of variables that we'll use in Resolve(), but override them with values from the actual environment in case there's any conflict.`  Happy to tweak the wording if that's omitting something.

Reply:
That looks good to me.  I was just concerned that a year from now I'd forget the importance of that ordering.